scripts/lint_style.lean

```diff
@@ -26,7 +26,9 @@ def lintStyleCli (args : Cli.Parsed) : IO UInt32 := do
   for s in ["Archive.lean", "Counterexamples.lean", "Mathlib.lean"] do
     let n ← lintAllFiles (System.mkFilePath [s]) errorStyle
     numberErrorFiles := numberErrorFiles + n
-  return numberErrorFiles
+  -- Make sure to return an exit code of at most 125, so this return value can be used further
+  -- in shell scripts.
+  return min numberErrorFiles 125
```

Comment:
```suggestion
  return min (numberErrorFiles % 128) 125
```
I wonder whether this can be (ever so slightly) more meaningful:
* reduce module 128=2^7 (since 126 and 127 are excluded for other reasons, but 128 appears to be the meaningful number);
* truncate at 125 to avoid 126 and 127.

Reply:
But this would mishandle the case of exactly 128 errors, right? I just pushed a commit allowing 128 explicitly - it is somewhat longer. Do you think returning an exit code of 128 is an important feature? I am not convinced the longer code is worth it, but could be swayed otherwise.

Reply:
Sorry, I got this wrong.  We should special-case *actual value* 0 and everything else.

How about something like
```lean
if numberErrorFiles == 0 then return 0 else
  return max 1 (numberErrorFiles % 125)
```

Reply:
This makes more sense! Why should we do *modulo* though, instead of saturating? I'd prefer 130 files to return a code of 125 over 2 (or 5), WDYT?

Reply:
Sure, I do not have a preference.  There might even be some value into specifying that exit code 1 means there were some errors and leave other error codes for potentially other meanings.  Feel free to pick what you prefer: let's simply make sure that exit code 0 is if and only if `numberErrorFiles = 0`!  :smile:

Reply:
I'll let a maintainer weigh in on their preference: I have a weak preference for simply reverting the most recent commit; I'm happy to do either of the other options.

Reply:
Saturation sounds better to me, but we're in the weeds here. :-)